src/main/java/com/iexec/blockchain/broker/BrokerService.java

```diff
@@ -52,28 +52,50 @@ public BrokerService(ChainConfig chainConfig, IexecHubService iexecHubService) {
                 .target(BrokerClient.class, chainConfig.getBrokerUrl());
     }
 
+    void checkAssetOrder(String requestOrderAddress, String assetAddress, BigInteger assetPrice, String assetType) {
+        Objects.requireNonNull(requestOrderAddress,  assetType + " address cannot be null in request order");
+        Objects.requireNonNull(assetAddress, assetType + " address cannot be null");
+        Objects.requireNonNull(assetPrice, assetType + " price cannot be null");
+        if (!requestOrderAddress.equalsIgnoreCase(assetAddress)) {
+            throw new IllegalStateException(assetType + " address is not the same in order and request order");
+        }
+    }
+
+    void checkRequestOrder(RequestOrder requestOrder) {
+        Objects.requireNonNull(requestOrder, "Request order cannot be null");
+        Objects.requireNonNull(requestOrder.getAppmaxprice(), "Requester application max price cannot be null");
+        Objects.requireNonNull(requestOrder.getWorkerpoolmaxprice(), "Requester worker pool max price cannot be null");
```

Comment:
```suggestion
        Objects.requireNonNull(requestOrder.getWorkerpoolmaxprice(), "Requester workerpool max price cannot be null");
```
I assume we don't usually put space between "worker" & "pool".